### PR TITLE
npm dev1

### PR DESCRIPTION
**Pull request by legobitna**
Hello @chug2k  i am doing the Issue you post 
but i got into to question
1. as you can see, inside of scripts, i made "dev" command and i put 2 commands one for server another for app, but when i run "npm run dev", it doesn't run the second command. 
2. I hardcode 3001 port number at the "dev" command. im not sure it's what you want. i try to use 'JSON_DEV_PORT' inside of command but it can't bring the value, but i use this value for react app (Jobs.js, Details.js) I'm not sure this is how you want

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "@testing-library/react": "^9.5.0",
     "@testing-library/user-event": "^7.2.1",
     "bootstrap": "^4.4.1",
+    "dotenv": "^8.2.0",
```
Comment (chug2k) [inline package.json:13]: this is incorrect. please move this to `devDependencies`. You should have run this -`-save-dev`

File: src/pages/Jobs.js [hunk 2]
```diff
@@ -19,8 +19,9 @@ export default function Jobs() {
   let [keyword, setKeyword] = useState(query.get(QUERYSTR_PREFIX));
 
   const getData = async () => {
-    //let url ="http://localhost:3001/jobs";
-    let url = `https://my-json-server.typicode.com/legobitna/Itviec/jobs`;
+    let url =`http://localhost:${process.env.REACT_APP_JSON_DEV_PORT}/jobs`;
+    //let url = process.env.REACT_APP_PRODUCTION_JSON_URL;
+    console.log("vu",url)
```
Comment (chug2k) [inline src/pages/Jobs.js:24]: remove

File: src/pages/Jobs.js [hunk 3]
```diff
@@ -19,8 +19,9 @@ export default function Jobs() {
   let [keyword, setKeyword] = useState(query.get(QUERYSTR_PREFIX));
 
   const getData = async () => {
-    //let url ="http://localhost:3001/jobs";
-    let url = `https://my-json-server.typicode.com/legobitna/Itviec/jobs`;
+    let url =`http://localhost:${process.env.REACT_APP_JSON_DEV_PORT}/jobs`;
```
Comment (chug2k) [inline src/pages/Jobs.js:22]: let url = `${process.env.BACKEND_URL}/jobs`

**Comment by chug2k** [comment]
Good questions. This article at https://dev.to/numtostr/environment-variables-in-node-js-the-right-way-15ad is a pretty good answer (with some good background reading), specifically from point 3. 

The port is actually annoying, I didn't think of that. It's okay to just hardcode 3001, let's not make a separate variable for that.

Essentially:
   - You always read from BACKEND_SERVER_URL
   - in .env, you set BACKEND_SERVER_URL=localhost:3001

On production (after you push), you set the environment variable BACKEND_SERVER_URL to be my-typicode.whatever.whatever.

File: package.json (hunk 1 above)
Comment (legobitna) [reply package.json:13]: solve it!

File: src/pages/Jobs.js (hunk 3 above)
Comment (legobitna) [reply src/pages/Jobs.js:22]: solved it

File: src/pages/Jobs.js (hunk 2 above)
Comment (legobitna) [reply src/pages/Jobs.js:24]: solved it

File: src/pages/Jobs.js (hunk 3 above)
Comment (legobitna) [reply src/pages/Jobs.js:22]: solved it

File: package.json [hunk 4]
```diff
@@ -22,7 +21,7 @@
     "redux": "^4.0.5"
   },
   "scripts": {
-    "dev": "json-server --watch db.json -p 3001 & react-scripts start",
+    "dev": "json-server --watch db.json -p 3001 && start react-scripts start",
```
Comment (chug2k) [inline package.json:24]: is this right? you need start?

**Comment by legobitna** [comment]
@chug2k  if you approve this code and readme file, i will merge it

File: package.json (hunk 4 above)
Comment (legobitna) [reply package.json:24]: yeah  we need start to start the second command

**Pull request merged**